vertexai/token-count/tokencount.go

```diff
@@ -0,0 +1,67 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// token-count shows an example of determining how many tokens correspond to a given prompt string
+package main
+
+// [START aiplatform_gemini_token_count]
+import (
+	"context"
+	"fmt"
+	"io"
+	"log"
+	"os"
+
+	"cloud.google.com/go/vertexai/genai"
+)
+
+func main() {
+	projectID := os.Getenv("GOOGLE_CLOUD_PROJECT")
+	location := "us-central1"
+	modelName := "gemini-pro"
+
+	prompt := "How many tokens are there in this prompt?"
```

Comment:
Nit: this example confused me a bit the first time I read it, because it seemed like you're asking the model about the token count, instead of calling a special API for this that returns a deterministic result w/o a LLM in the way.

IMHO making it something like:

```
prompt := "a sample prompt"
```

While more boring - has a smaller chance to confound

Reply:
Makes sense!

We'd want:
- to use the same sample prompt in Go, Java, etc.
- a string that would make a meaningful prompt e.g. "How can I change a lightbulb?"

Reply:
Changed to "why is sky blue?", taken from the Python sample